src/pages/home/report/ReportView.js

```diff
@@ -4,40 +4,53 @@ import PropTypes from 'prop-types';
 import ReportActionsView from './ReportActionsView';
 import ReportActionCompose from './ReportActionCompose';
 import {addAction} from '../../../libs/actions/Report';
+import compose from '../../../libs/compose';
 import KeyboardSpacer from '../../../components/KeyboardSpacer';
 import styles from '../../../styles/styles';
 import SwipeableView from '../../../components/SwipeableView';
+import FullScreenLoadingIndicator from '../../../components/Loading/FullscreenLoading';
+import withDrawerState from '../../../components/withDrawerState';
+import withWindowDimensions from '../../../components/withWindowDimensions';
 
 const propTypes = {
-    // The ID of the report actions will be created for
+    /* The ID of the report the selected report */
     reportID: PropTypes.number.isRequired,
+
+    /* Is the view ready to be displayed */
+    loaded: PropTypes.bool.isRequired,
+
+    /* Is the report view covered by the drawer */
+    isDrawerOpen: PropTypes.bool.isRequired,
+
+    /* Is the window width narrow, like on a mobile device */
+    isSmallScreenWidth: PropTypes.bool.isRequired,
 };
 
-// This is a PureComponent so that it only re-renders when the reportID changes or when the report changes from
-// active to inactive (or vice versa). This should greatly reduce how often comments are re-rendered.
-class ReportView extends React.Component {
-    shouldComponentUpdate(prevProps) {
-        return this.props.reportID !== prevProps.reportID;
-    }
-
-    render() {
-        return (
-            <View style={[styles.chatContent]}>
-                <ReportActionsView
-                    reportID={this.props.reportID}
-                />
+function ReportView(props) {
+    const isDrawerOutOfTheWay = !props.isDrawerOpen || !props.isSmallScreenWidth;
+
+    return (
+        <View key={props.reportID} style={[styles.chatContent]}>
+            {!props.loaded && <FullScreenLoadingIndicator />}
+
+            <ReportActionsView reportID={props.reportID} />
+
+            {isDrawerOutOfTheWay && (
                 <SwipeableView onSwipeDown={() => Keyboard.dismiss()}>
                     <ReportActionCompose
-                        onSubmit={text => addAction(this.props.reportID, text)}
-                        reportID={this.props.reportID}
-                        key={this.props.reportID}
+                        onSubmit={text => addAction(props.reportID, text)}
+                        reportID={props.reportID}
+                        autoFocus={!props.isSmallScreenWidth}
                     />
                 </SwipeableView>
-                <KeyboardSpacer />
-            </View>
-        );
-    }
+            )}
```

Comment:
I've just spotted an Android issue coming from this: 
1. Open a chat
2. Focus the input field
3. Press the back arrow to go back to the chat list
You'll get this error: 

I'll debug it tomorrow. Worst case we can leave the field mounted...